### PR TITLE
Bugfix 282 image picker autocomplete

### LINKED ISSUE #282: image picker, autocomplete fixed

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #282

### CODE REVIEW

File: src/components/pages/WelcomePage.tsx [hunk 1]
```diff
@@ -682,6 +704,67 @@ class WelcomePage extends React.Component<Props, State> {
...
+
+      return (
+        <Autocomplete<AutocompleteItem>
+          options={ autocompleteOptions }
+          inputValue={ autocompleteInput }
+          onInputChange={ this.onAutocompleteInputChange }
+          value={ autocompleteValue }
+          getOptionLabel={ option => option.label }
+          onChange={ this.onAutocompleteChange }
+          renderInput={(params) =>
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:727]:
```suggestion
          renderInput={ params =>
```

File: src/components/pages/WelcomePage.tsx [hunk 2]
```diff
@@ -682,6 +704,67 @@ class WelcomePage extends React.Component<Props, State> {
...
+            <TextField
+              style={{ border: "0.5px rgba(0,0,0,0.5) solid" }}
+              variant="outlined"
+              {...params}
+              InputProps={{ ...params.InputProps }}
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:732]:
```suggestion
              InputProps={{ ...params.InputProps }}
              { ...params }
```

File: src/components/pages/WelcomePage.tsx [hunk 3]
```diff
@@ -682,6 +704,67 @@ class WelcomePage extends React.Component<Props, State> {
...
+   * Event handler for autocomplete input change
+   * 
+   * @param _event event
+   * @param newInputValue new input value
+   */
+  private onAutocompleteInputChange = (_event: React.ChangeEvent<{}>, newInputValue: string) => {
+    this.setState({
+      autocompleteInput: newInputValue
+    });
+
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:750]:
```suggestion
    this.setState({ autocompleteInput: newInputValue });
```

File: src/components/pages/WelcomePage.tsx [hunk 4]
```diff
@@ -682,6 +704,67 @@ class WelcomePage extends React.Component<Props, State> {
...
+   * 
+   * @param _event event
+   * @param value new value
+   */
+  private onAutocompleteChange = (_event: React.ChangeEvent<{}>, value: AutocompleteItem | null) => {
+    const { formValues } = this.state;
+    this.setFieldValue("location", value?.value || null);
+    this.setState({
+      autocompleteValue: value || undefined,
+    });
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:764]:
```suggestion
    this.setState({ autocompleteValue: value || undefined });
```

File: src/components/pages/WelcomePage.tsx [hunk 5]
```diff
@@ -983,40 +1066,25 @@ class WelcomePage extends React.Component<Props, State> {
...
-      this.setState({
-        fetchData: data,
-        fethingData: false
-      });
-      await this.returnPlaces();
-    }
-    return this.returnPlaces();
+    this.setState({
+      autocompleteOptions: autocompleteOptions
+    })
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:1076]:
```suggestion
    this.setState({ autocompleteOptions: autocompleteOptions })
```

File: src/components/pages/image-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      <div className={ classes.imagePicker }>
+        { images.map(image => this.renderImage(image, () => onPick(image))) }
+        <div className={ classes.imagePickerSelectedClear }/>
+      </div>
+    );
+  } 
+
+}
+
+export default withStyles(styles)(ImagePicker)
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:45]:
```suggestion
export default withStyles(styles)(ImagePicker);
```

File: src/components/pages/image-picker.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        onClick={ onImagePick }>
+        <img src={image.src}
+          className={classNames(classes.imagePickerResponsiveThumbnail, image.selected && classes.imagePickerSelected)}
+          style={{ width: 150, height: 150, objectFit: "cover" }}
+        />
+      </div>
+    );
+  }
+
+  public render = () => {
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:32]:
```suggestion
  /**
   * Component render method
   */
  public render = () => {
```

File: src/components/pages/image-picker.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  images: Image[];
+  onPick: (image: Image) => void
+}
+
+class ImagePicker extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props)
+  }
+
+  private renderImage = (image: Image, onImagePick: () => void) => {
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:18]:
```suggestion
  /**
   * Renders image
   *
   * @param image image
   * @param onImagePick on image pick event handler
   */
  private renderImage = (image: Image, onImagePick: () => void) => {
```

File: src/components/pages/image-picker.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import { Done } from '@material-ui/icons';
+import styles from "../../styles/image-picker";
+import classNames from 'classnames';
+
+interface Props extends WithStyles<typeof styles> {
+  images: Image[];
+  onPick: (image: Image) => void
+}
+
+class ImagePicker extends React.Component<Props> {
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:13]:
```suggestion
/**
 * Image picker component
 */
class ImagePicker extends React.Component<Props> {
```

File: src/components/pages/image-picker.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { Image } from "../../types/image-picker";
+import { WithStyles, withStyles } from '@material-ui/core';
+import { Done } from '@material-ui/icons';
+import styles from "../../styles/image-picker";
+import classNames from 'classnames';
+
+interface Props extends WithStyles<typeof styles> {
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:8]:
```suggestion
/**
 * Component properties
 */
interface Props extends WithStyles<typeof styles> {
```

File: src/components/pages/image-picker.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+    return (
+      <div className={classNames(classes.imagePickerResponsive, image.selected && classes.imagePickerResponsiveSelected)}
+        onClick={ onImagePick }>
+        <img src={image.src}
+          className={classNames(classes.imagePickerResponsiveThumbnail, image.selected && classes.imagePickerSelected)}
+          style={{ width: 150, height: 150, objectFit: "cover" }}
+        />
+      </div>
+    );
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:29]:
```suggestion
    return (
      <div className={ classNames(classes.imagePickerResponsive, image.selected && classes.imagePickerResponsiveSelected) }
        onClick={ onImagePick }>
        <img
          src={ image.src }
          className={ classNames(classes.imagePickerResponsiveThumbnail, image.selected && classes.imagePickerSelected) }
          style={{ width: 150, height: 150, objectFit: "cover" }}
        />
      </div>
    );
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/pages/image-picker.tsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+  /**
+   * Component render method
+   */
+  /**
+   * Component render method
+   */
+  /**
+   * Component render method
+   */
```
Comment (Chi-Chu319) [inline src/components/pages/image-picker.tsx:53]:
```suggestion
  /**
   * Component render method
   */
```

**Review by Chi-Chu319 — COMMENTED** [review]
removed duplicated

File: src/components/pages/WelcomePage.tsx [hunk 13]
```diff
@@ -585,7 +604,7 @@ class WelcomePage extends React.Component<Props, State> {
             style={ this.state.showDefaultImages && !imageUrl ? {display:"block"} : {display:"none"} }
           >
             <ImagePicker
-              images={ imageList.map((image, i) => ({src: image, value: i})) }
+              images={ imageList.map((image, i) => ({src: image, value: i, selected: defaultImageUrl === image})) }
```
Comment (lauriLukkarinen) [inline src/components/pages/WelcomePage.tsx:607]:
```suggestion
              images={ imageList.map((image, i) => ({ src: image, value: i, selected: defaultImageUrl === image })) }
```

File: src/components/pages/image-picker.tsx [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+  images: Image[];
+  onPick: (image: Image) => void
+}
+
+/**
+ * Image picker component
+ */
+class ImagePicker extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props)
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:21]:
```suggestion
    super(props);
```

File: src/components/pages/image-picker.tsx [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+   * Renders image
+   *
+   * @param image image
+   * @param onImagePick on image pick event handler
+   */
+  private renderImage = (image: Image, onImagePick: () => void) => {
+    const { classes } = this.props;
+
+    return (
+      <div className={ classNames(classes.imagePickerResponsive, image.selected && classes.imagePickerResponsiveSelected) }
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:34]:
```suggestion
      <div className={ classNames(classes.imagePickerResponsive, { [classes.imagePickerResponsiveSelected]: image.selected }) }
```

File: src/components/pages/image-picker.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+        onClick={ onImagePick }>
+        <img
+          src={ image.src }
+          className={ classNames(classes.imagePickerResponsiveThumbnail, image.selected && classes.imagePickerSelected) }
```
Comment (lauriLukkarinen) [inline src/components/pages/image-picker.tsx:38]:
```suggestion
          className={ classNames(classes.imagePickerResponsiveThumbnail, { [classes.imagePickerSelected]: image.selected }) }
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**